rclpy/rclpy/node.py

```diff
@@ -761,6 +763,36 @@ def _set_parameters_atomically(
 
         return result
 
+    def add_on_set_parameters_callback(
+        self,
+        callback: Callable[[List[Parameter]], SetParametersResult]
+    ) -> None:
+        """
+        Add a callback in front to the list of callback.
+
+        Calling this function will add a callback in self._parameter_callback list.
+
+        :param callback: The function that is called whenever parameters are set for the node.
+        """
+        self._parameters_callback.insert(0, callback)
+
+    def remove_on_set_parameters_callback(
+        self,
+        callback: Callable[[List[Parameter]], SetParametersResult]
+    ) -> None:
+        """
+        Remove a callback from list of callback.
+
+        Calling this function will remove the callback from self._parameter_callback list.
+
+        :param callback: The function that is called whenever parameters are set for the node.
+        :raises: ValueError if a callback is not present in the list of callback.
+        """
+        try:
+            self._parameters_callback.remove(callback)
+        except ValueError as vr:
+            print(vr)
```

Comment:
This doesn't raise an exception, this just prints the exception. To allow it to propagate up the stack, remove the `try` and `except` statements, and just leave this as `self._parameters_callback.remove(callback)`.